standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/dataconnector/jdbc/AbstractJDBCConnectorProvider.java

```diff
@@ -129,7 +129,30 @@ protected Connection getConnection() {
    * @throws MetaException To indicate any failures with executing this API
    * @param regex
    */
-  @Override public abstract List<Table> getTables(String regex) throws MetaException;
+  @Override public List<Table> getTables(String regex) throws MetaException {
+    ResultSet rs = null;
+    try {
+      rs = fetchTablesViaDBMetaData(regex);
+      if (rs != null) {
+        List<Table> tables = new ArrayList<Table>();
+        while(rs.next()) {
+          tables.add(getTable(rs.getString(3)));
```

Comment:
so getTable() can throw an exception when retrieving column info for each table. What should happen when there is an exception on a certain table? Currently, we are ending the iteration and throwing an exception. I dont know what the right behavior should be, whether to return the remaining table or to throw an exception. Do you know what the JDBC spec/javadocs for getTables() says/does for this?

Reply:
If I am looking at the right [place](https://javadoc.scijava.org/Java6/java/sql/DatabaseMetaData.html). It does not specify its behavior neither. But I think it makes more sense to just leave out the one with column exception and continue returning the rest instead of stopping? Because we want the behavior to be as close as getTables(), we can just filter out the ones that has corrupted columns and return as many as we can.